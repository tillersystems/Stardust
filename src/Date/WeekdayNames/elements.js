import styled from 'styled-components';

export const DayName = styled.div`
  font-size: ${({
    theme: {
      fonts: { size },
    },
  }) => size.medium};
`;
