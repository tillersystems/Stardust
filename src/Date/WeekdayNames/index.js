import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { Info } from 'luxon';

import { DayName } from './elements';

/**
 * Week day Names
 *
 * This component is in charge of displaying
 * the week day names
 *
 * @param {string} className // Class needed by styled component.
 *
 * @return {jsx}
 */

const WeekdayNames = ({ className, locale }) => {
  const dayNames = Info.weekdays('short', { locale });
  return (
    <div className={className} key={name}>
      {dayNames.map(dayName => (
        <DayName key={dayName}>{dayName}</DayName>
      ))}
    </div>
  );
};

/**
 * PropTypes Validation.
 */
const { string } = PropTypes;
WeekdayNames.propTypes = {
  className: string,
  locale: string,
};

/**
 * Default props.
 */
WeekdayNames.defaultProps = {
  className: '',
  locale: 'en',
};

export default styled(WeekdayNames)`
  display: grid;
  grid-template-columns: repeat(7, 2.8rem);
  grid-template-rows: 2rem;
  justify-items: stretch;
  align-items: stretch;

  text-align: center;

  border-bottom: 1px solid ${({ theme: { palette } }) => palette.mediumGrey};

  color: ${({ theme: { palette } }) => palette.mediumGrey};
`;
