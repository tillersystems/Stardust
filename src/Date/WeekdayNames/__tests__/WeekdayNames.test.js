import React from 'react';

import WeekdayNames from '..';

describe('<WeekdayNames />', () => {
  test('should render withouth a problem', () => {
    const { container } = render(<WeekdayNames />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render with a different locale', () => {
    const { container } = render(<WeekdayNames locale="ar" />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
