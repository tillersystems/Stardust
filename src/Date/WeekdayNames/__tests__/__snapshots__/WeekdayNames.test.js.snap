// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeekdayNames /> should render with a different locale 1`] = `
.c1 {
  font-size: 1.4rem;
}

.c0 {
  display: grid;
  grid-template-columns: repeat(7,2.8rem);
  grid-template-rows: 2rem;
  justify-items: stretch;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  text-align: center;
  border-bottom: 1px solid hsl(206,23%,69%);
  color: hsl(206,23%,69%);
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    Mon
  </div>
  <div
    class="c1"
  >
    Tue
  </div>
  <div
    class="c1"
  >
    Wed
  </div>
  <div
    class="c1"
  >
    Thu
  </div>
  <div
    class="c1"
  >
    Fri
  </div>
  <div
    class="c1"
  >
    Sat
  </div>
  <div
    class="c1"
  >
    Sun
  </div>
</div>
`;

exports[`<WeekdayNames /> should render withouth a problem 1`] = `
.c1 {
  font-size: 1.4rem;
}

.c0 {
  display: grid;
  grid-template-columns: repeat(7,2.8rem);
  grid-template-rows: 2rem;
  justify-items: stretch;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  text-align: center;
  border-bottom: 1px solid hsl(206,23%,69%);
  color: hsl(206,23%,69%);
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    Mon
  </div>
  <div
    class="c1"
  >
    Tue
  </div>
  <div
    class="c1"
  >
    Wed
  </div>
  <div
    class="c1"
  >
    Thu
  </div>
  <div
    class="c1"
  >
    Fri
  </div>
  <div
    class="c1"
  >
    Sat
  </div>
  <div
    class="c1"
  >
    Sun
  </div>
</div>
`;
